src/wallet/init.cpp

```diff
@@ -85,6 +87,7 @@ void WalletInit::AddWalletOptions(ArgsManager& argsman) const
 
     argsman.AddArg("-hdseed=<hex>", "User defined seed for HD wallet (should be in hex). Only has effect during wallet creation/first start (default: randomly generated)", ArgsManager::ALLOW_ANY | ArgsManager::SENSITIVE, OptionsCategory::WALLET_HD);
     argsman.AddArg("-mnemonic=<text>", "User defined mnemonic for HD wallet (bip39). Only has effect during wallet creation/first start (default: randomly generated)", ArgsManager::ALLOW_ANY | ArgsManager::SENSITIVE, OptionsCategory::WALLET_HD);
+    argsman.AddArg("-mnemonicbits=<n>", strprintf("User defined mnemonic security for HD wallet in bits (BIP39). Only has effect during wallet creation/first start (allowed values: 128, 192, 256; default: %u)", CHDChain::DEFAULT_MNEMONIC_BITS), ArgsManager::ALLOW_ANY, OptionsCategory::WALLET_HD);
```

Comment:
isn't 160 also accepted as a correct value?

```
    // check number of words
    if (nWordCount != 12 && nWordCount != 18 && nWordCount != 24) {
        return false;
    }
```

but when it's generated:
```
    if (strength % 32 || strength < 128 || strength > 256) {
```

Probably should be updated `32` to 64: otherwise Generate() is happy with 160 bits, 224 bits but `CMnemonic::Check` later fails with error: mnemonic with 15 words are invalid mnemonic.

Reply:
I tested it with modified script:

```
--- a/test/functional/wallet_mnemonicbits.py
+++ b/test/functional/wallet_mnemonicbits.py
@@ -28,6 +28,8 @@ class WalletMnemonicbitsTest(BitcoinTestFramework):
         assert_equal(len(self.nodes[0].dumphdinfo()["mnemonic"].split()), 12)  # 12 words by default
 
         self.log.info("Can have multiple wallets with different mnemonic length loaded at the same time")
+        self.restart_node(0, extra_args=self.extra_args[0] + ["-mnemonicbits=160"])
+        self.nodes[0].createwallet("wallet_160")
         self.restart_node(0, extra_args=self.extra_args[0] + ["-mnemonicbits=192"])
         self.nodes[0].createwallet("wallet_192")
         self.restart_node(0, extra_args=self.extra_args[0] + ["-mnemonicbits=256"])
@@ -35,6 +37,7 @@ class WalletMnemonicbitsTest(BitcoinTestFramework):
         self.nodes[0].createwallet("wallet_256", False, True)  # blank
         self.nodes[0].get_wallet_rpc("wallet_256").upgradetohd()
         assert_equal(len(self.nodes[0].get_wallet_rpc(self.default_wallet_name).dumphdinfo()["mnemonic"].split()), 12)  # 12 words by default
+        assert_equal(len(self.nodes[0].get_wallet_rpc("wallet_160").dumphdinfo()["mnemonic"].split()), 15)              # 15 words
```
...
```
test_framework.authproxy.JSONRPCException: SetMnemonic: invalid mnemonic: `noise glow drift unveil intact fruit mind add claim ankle pumpkin cheap come tuna win` (-1)
```

Reply:
Good catch! Indeed more values should be allowed https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki#generating-the-mnemonic